client/network/src/protocol/message.rs

```diff
@@ -137,14 +137,72 @@ pub struct RemoteReadResponse {
 
 /// Generic types.
 pub mod generic {
+	use bitflags::bitflags;
 	use codec::{Encode, Decode, Input, Output};
 	use sp_runtime::Justification;
-	use crate::config::Roles;
 	use super::{
 		RemoteReadResponse, Transactions, Direction,
 		RequestId, BlockAttributes, RemoteCallResponse, ConsensusEngineId,
 		BlockState, StorageProof,
 	};
+
+	bitflags! {
```

Comment:
We are serializing the new `enum Role` manually, so I don't think any of this is needed.

Reply:
We're still passing around `Rules` in the internals of `sc_network`, so I'd leave them. I'll remove the `TODO` though.